### ISSUE TITLE
SelectOption.value and Select.selected has "any" type, but Select.onSelect has "(..., string | number)"

### ISSUE THREAD

**Issue by mvollmer** [patternfly/patternfly-react#11361]
When looking at the `SelectOption.value` and `Select.selected` properties, I got the impression that one can use any JavaScript value to identify select options, and it indeed works as expected.

But it doesn't type-check: The `Select.onSelect` property wants a function that takes `string | number`. I had expected it to also use `any`.

We have our own variant of your SimpleSelect template, and we would like to make it generic in the type of the option value. This seems to work, except for this little niggle about the type of `Select.onSelect`.

**Comment by mvollmer** [comment]
This is our attempt at making SimpleSelect and CheckboxSelect generic: https://github.com/cockpit-project/cockpit/pull/21454

**Comment by wise-king-sullyman** [comment]
I agree that the types are conflicting and that's something I do think we should fix, so thank you for making this issue!

I am a bit confused though as to what drives the need to pass a generic type for the `SelectOption.value` though. Since it's effectively just an identifier what's the use case for non string/number values?
